### ISSUE TITLE
Doctests as part of travis

### ISSUE THREAD

**Issue by vchuravy** [JuliaLang/julia#15815]
Doctests are part of the documentation and having them run automatically on travis, would ensure that the Julia documentation is kept up to date and consistent. Additionally more tests are always helpful and #15753 exposed a few regressions.

After #15753 all doctests are functional again and we could enable them for testing. While working on that PR I noticed that there is one annoying property of the doctest. They may spuriously fail when linenumbers change. This can be solved by replacing the linenumbers in the doctest with `...`. Any other changes will still trigger a failure.

As an example see https://github.com/JuliaLang/julia/pull/15753/commits/19d44318d14e4a880f3eaedf6c29f4fdf310c83a

**Comment by tkelman** [comment]
This will be yet another way in which travis can fail if people push changes straight to master without testing first, but I think it would be worth it.

For simple stuff like line numbers, I wonder if we couldn't make JuliaDoc fill them in automatically? This may be easier for the doctests that have already been migrated to inline julia docstrings.

**Comment by yuyichao** [comment]
It'll be nice if the backtrace can be ignored .... As we improve the backtrace for inline functions the backtrace is likely going to be more and more sensitive to subtle changes that can affect line numbers or inlining decisions.

**Comment by vchuravy** [comment]
We probably could make JuliaDoc less sensitive to line-number changes, or alternativly we encourage the usage of `...`. As far as I can tell it is very sparsily used in the doctests. One example of where it is well used is: https://github.com/JuliaLang/julia/commit/adce3569b31d29e8bf45c3b8f66fdf2dbd2ec947#diff-c9107d78b307be1cab2363ee98efe06bR27.

**Comment by tkelman** [comment]
I actually like having a full detailed backtrace for release docs, but it is overkill while in dev.

**Comment by omus** [comment]
While working on #17106 I discovered some good reasons not to run doctests in dev (release may still be fine):

The doctest `julia> methods(+)` would sometimes print the methods in a different order resulting in the doctest to fail.

If we were to run the doctests as part of the Travis we would have to limit them to run on only 64-bit as integer literals reveal their types in exceptions :
```julia
julia> bar(a,b,x::Vararg{Any,2}) = (a,b,x);

julia> bar(1,2,3)
ERROR: MethodError: no method matching bar(::Int64, ::Int64, ::Int64)
...
```
I would however like to see some automation with the doctest system to ensure that the manual is kept up to date.

**Comment by KristofferC** [comment]
Closing in favor of https://github.com/JuliaLang/julia/issues/15815

**Comment by tkelman** [comment]
this is 15815

**Comment by KristofferC** [comment]
Derp, https://github.com/JuliaLang/julia/issues/19528.
